### DISCUSSION TITLE
How to use socket.io to more than one module in nodejs

### DISCUSSION THREAD

**Discussion by sbn111** [socketio/socket.io#4578] [Q&A]
Hi everyone,

I have configured socket.io in server.js file and its working perfectly

Now I want to emit and send client to some data from a router file.

Can anyone pls tell me how to achieve that?

**Comment by verbose117** [comment]
Could you provide the context you're working with?

If you're looking to reuse a socket. Here's how I did it:

```js
export default function socket(httpServer) {
  const io = new Server(httpServer);

  io.on("connection", socket => {
    // handle connection
  });

  return io;
}
```

```js
const server = require("http").createServer();
const io = socket(server);

export { io };
```
